File: RETOS/Reto30/Reto30.py
def encriptar_mensaje(mensaje, clave):
    mensaje_encriptado = ""
    clave = clave % 26  # Asegurar que la clave sea dentro de 0-25

    for caracter in mensaje:
        # Encriptar solo letras
        if caracter.isalpha():
            base = 'a' if caracter.islower() else 'A'
            caracter = chr((ord(caracter) - ord(base) + clave) % 26 + ord(base))

        mensaje_encriptado += caracter

    return mensaje_encriptado


# Solicitar al usuario el mensaje y la clave
mensaje = input("Introduce el mensaje: ")
clave = int(input("Introduce la clave: "))

# Encriptar el mensaje
mensaje_encriptado = encriptar_mensaje(mensaje, clave)

# Mostrar el mensaje encriptado
print("Mensaje encriptado:", mensaje_encriptado)

File: RETOS/Reto30/Reto30.js
const readline = require('readline');


const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function encriptarMensaje(mensaje, clave) {
    let mensajeEncriptado = '';
    clave = clave % 26; 

    for (let i = 0; i < mensaje.length; i++) {
        let caracter = mensaje[i];
        if (/[a-zA-Z]/.test(caracter)) {
            let base = (caracter === caracter.toLowerCase()) ? 'a' : 'A';
            caracter = String.fromCharCode(base.charCodeAt(0) + (caracter.charCodeAt(0) - base.charCodeAt(0) + clave) % 26);
        }
        mensajeEncriptado += caracter;
    }

    return mensajeEncriptado;
}


rl.question('Introduce el mensaje: ', (mensaje) => {
    rl.question('Introduce la clave: ', (claveStr) => {
        let clave = parseInt(claveStr);
        let mensajeEncriptado = encriptarMensaje(mensaje, clave);
        console.log("Mensaje encriptado: " + mensajeEncriptado);
        rl.close();
    });
});
